### PR TITLE
Add howto doc for salesforce connection

### LINKED ISSUE #10029: Add Salesforce docs for setting up a connection
**Description**

As of the related issue on adding a custom connection (see issue link below) we need a how-to guide on setting up a connection to salesforce.

To be more specific we need to add a `salesforce.rst` to the `docs/howto/connection` directory explaining the fields you need to set to setup a proper connection to salesforce. A connection will be established through the `airflow/providers/salesforce/hooks/salesforce.py` hook. Check out this file to see how the connection is established and describe this in the guide you are adding.
You can check other how-to connection guides in the directory to see how it works. If you have further questions do not hesitate to ask. :)

**Use case / motivation**

If you simply want to connect to salesforce services you should be able to do so without investigating the code and instead just read a proper howto docs explaining what to do.

**Related Issues**

https://github.com/apache/airflow/issues/8766

### PR DESCRIPTION
**Pull request by eladkal**
closes: #10029
related: #8766

This PR adds howto guide for setting up Salesforce connection with Airflow

### CODE REVIEW

File: docs/howto/connection/salesforce.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+.. code-block:: json
+
+    {
+      "security_token": "your_token",
+      "domain":"test"
+    }
+
+.. note::
+  Airflow currently does not support other login methods such as IP filtering and JWT.
```
Comment (mik-laj) [inline docs/howto/connection/salesforce.rst:73]: Can you add example URI? This is helpful if you configure the connection using environment variables or secrets backend.

File: docs/howto/connection/salesforce.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+.. _howto/connection:SalesforceHook:
+
+Salesforce Connection
+==============================
```
Comment (feluelle) [inline docs/howto/connection/salesforce.rst]:
```suggestion
Salesforce Connection
=====================
```

File: docs/howto/connection/salesforce.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    under the License.
+
+.. _howto/connection:SalesforceHook:
+
+Salesforce Connection
+=====================
+The HTTP connection type provides connection to Salesforce.
+
+Configuring the Connection
+==========================
```
Comment (mik-laj) [inline docs/howto/connection/salesforce.rst]: The page should only have one title. Currently, due to bad headings, there are several of them.

Comment (eladkal) [reply docs/howto/connection/salesforce.rst]: @mik-laj @feluelle fixed.
Should be OK now

**Review by mik-laj — APPROVED** [review]

**Comment by mik-laj** [comment]
> :x: CI Build / Build docs (pull_request) Failing after 8m — Build docs
> :x: CI Build / Spell check docs (pull_request) Failing after 5m — Spell check docs

Can you fix it?

**Comment by eladkal** [comment]
@mik-laj @feluelle rebased

**Review by mik-laj — APPROVED** [review]

**Pull request merged**